frontend/src/components/utils.js

```diff
@@ -52,3 +53,59 @@ export const pivotHeatMapData = (payload) => {
   });
   return allRubrics;
 };
+
+export const pivotHeatMapDataWithSections = (payload, sections) => {
+  // this transformation takes into account that not all criteria have the same rating set
+
+  // create a student lookup for student_id(key) and section_id(value) for later use
+  const studentLookup = Object.assign(...payload.denormalized_data.map((dp) => {
+    const { student_id: studentId, section_id: sectionId } = dp;
+    // apparrently the [] is needed around a computed value to assign object keys
+    return { [studentId]: sectionId };
```

Comment:
Yeah, good point. I try as often as I can to avoid the _Initialize Then Modify_ pattern, which I think usually bears good results. But your block is so much more readable than the hoops I had to jump through to get my implementation to work.

I think I might include my block as a "Don't Go Nuts with ES6" section of the ES6 Tech Talk. Is it ok if I include your block as an example of why readability should carry the most weight.

Reply:
Yeah absolutely. Love that idea. I suppose the block above could use `reduce()` which would avoid the initialization step. Old habits die hard, I guess :)